### ISSUE TITLE
'Invalid "auth" argument: {auth!r}' error for starlette & fastapi demo

### ISSUE THREAD

**Issue by ZhengRui** [authlib/demo-oauth-client#14]
Hi thanks for the demos, I tried some demos, the demo for flask works flawlessly, but when I tried the demos for fastapi & starlette (all google demos), it throws the error:
```bash

File "./app.py", line 40, in login
    return await oauth.google.authorize_redirect(request, redirect_uri)
  File "/usr/local/lib/python3.7/site-packages/authlib/integrations/starlette_client/integration.py", line 52, in authorize_redirect
    rv = await self.create_authorization_url(redirect_uri, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/authlib/integrations/base_client/async_app.py", line 51, in create_authorization_url
    metadata = await self.load_server_metadata()
  File "/usr/local/lib/python3.7/site-packages/authlib/integrations/base_client/async_app.py", line 20, in load_server_metadata
    metadata = await self._fetch_server_metadata(self._server_metadata_url)
  File "/usr/local/lib/python3.7/site-packages/authlib/integrations/base_client/async_app.py", line 204, in _fetch_server_metadata
    resp = await client.request('GET', url, withhold_token=True)
  File "/usr/local/lib/python3.7/site-packages/authlib/integrations/httpx_client/oauth2_client.py", line 95, in request
    method, url, auth=auth, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/httpx/_client.py", line 1482, in request
    request, auth=auth, allow_redirects=allow_redirects, timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/httpx/_client.py", line 1566, in send
    auth = self._build_request_auth(request, auth)
  File "/usr/local/lib/python3.7/site-packages/httpx/_client.py", line 433, in _build_request_auth
    self._auth if isinstance(auth, UseClientDefault) else self._build_auth(auth)
  File "/usr/local/lib/python3.7/site-packages/httpx/_client.py", line 425, in _build_auth
    raise TypeError(f'Invalid "auth" argument: {auth!r}')
TypeError: Invalid "auth" argument: <httpx._config.UnsetType object at 0x10a790950>
```

In `authlib/integrations/base_client/async_app.py` line 204
```python
    async def _fetch_server_metadata(self, url):
        async with self._get_oauth_client() as client:
            resp = await client.request('GET', url, withhold_token=True)
            return resp.json()
```
`withhold_token` is set to True, but didn't set `auth`, which cases `auth` to be `UNSET` in `authlib/integrations/httpx_client/oauth2_client.py` line 95 
```python
    async def request(self, method, url, withhold_token=False, auth=UNSET, **kwargs):
        if not withhold_token and auth is UNSET:
            if not self.token:
                raise MissingTokenError()

            if self.token.is_expired():
                await self.ensure_active_token()

            auth = self.token_auth

        return await super(AsyncOAuth2Client, self).request(
            method, url, auth=auth, **kwargs)
```
and throw the above error. 

My authlib version is:
```plain
google-auth-oauthlib==0.4.1
oauthlib==2.1.0
requests-oauthlib==1.0.0
httpx==0.18.2
```
Any fix for this ?

**Comment by lepture** [comment]
Please use httpx<0.18.2

**Comment by kujiy** [comment]
Fix for httpx 0.18.2
https://github.com/lepture/authlib/commit/40276708374fcc7c0ee6b1ce4955dbe7db82f5ba
